templates/keepalived.conf.j2

```diff
@@ -170,8 +170,27 @@ virtual_server {{ vserver.ip }} {{ vserver.port }} {
     {% if rserver.tcp_checks is defined %}
     {% for tcp_check in rserver.tcp_checks %}
     TCP_CHECK {
+      {% if tcp_check.connect_ip is defined and tcp_check.connect_ip %}
+      connect_ip {{ tcp_check.connect_ip }}
+      {% endif %}
+      {% if tcp_check.connect_port is defined and tcp_check.connect_port %}
       connect_port {{ tcp_check.connect_port }}
+      {% endif %}
       connect_timeout {{ tcp_check.connect_timeout | default('5') }}
+      retry {{ tcp_check.retry | default('1') }}
```

Comment:
Is retry necessary nowadays? I thought it wasn't. In that case I would prefer if we wrap it with a condition. It allows us to not vendor-in a default which could effectively change upstream.

Reply:
at least I've used it in the past :relaxed:
but I think it's a good idea to wrap these optional settings.
I will do that for all my newly added settings, but leave out the current available ones like
https://github.com/evrardjp/ansible-keepalived/blob/9e7a079abca9b0e2381f10e0dab4a0d66f5e8b35/templates/keepalived.conf.j2#L173-L174

https://github.com/evrardjp/ansible-keepalived/blob/9e7a079abca9b0e2381f10e0dab4a0d66f5e8b35/templates/keepalived.conf.j2#L199-L201

because of https://github.com/evrardjp/ansible-keepalived/pull/125#discussion_r365107536

Reply:
Thanks! That's indeed what I would hope.
Sorry for being a little bit picky, but I do what I think is for everyone's best :)

Reply:
NP. your code, your rules ;)